### ISSUE TITLE
asciidoctor writer currently needs to be fiddlier when considering escaping

### ISSUE THREAD

**Issue by rwalkerands** [jgm/pandoc#5488]
(pandoc 2.7.2)

I'm reporting this "for the record"; I think this is properly an issue to be addressed in asciidoctor, not pandoc, but it seems it may not be addressed very soon.

Due to an "interesting" design/implementation decision in asciidoctor, it's necessary for pandoc's asciidoctor writer to be fiddlier when deciding whether or not to output backslash escaping.

Please see https://github.com/asciidoctor/asciidoctor/issues/3297, where I reported an inconsistency with asciidoctor's handling of backslash-escaped braces. It turns out this was already reported as https://github.com/asciidoctor/asciidoctor/issues/901, and it's a more general issue than escaped braces.

In my case, given Markdown input:

```
Test escape {hello} and {} and { } and { a}.
```

the command

```
pandoc -f markdown -t asciidoctor t-backslash.md
```

outputs:
```
Test escape \{hello} and \{} and \{ } and \{ a}.
```

which I thought was completely reasonable.

However, it seems that asciidoctor takes the attitude "if I don't require a backslash to escape, don't give me one, because if I don't require one, but you give me one anyway, I'll pass through that backslash".

So if you run asciidoctor on pandoc's output to produce HTML, the result contains:
```
<p>Test escape {hello} and \{} and \{ } and \{ a}.</p>
```

Ugh.  So to keep asciidoctor happy, pandoc would have to output this instead:
```
Test escape \{hello} and {} and { } and { a}.
```

I say "currently" in the issue title, because it seems from the comments on the two asciidoctor issues that there is at least a possibility that asciidoctor could change its behaviour in future. And if that should happen, you'd then have to change pandoc back ...

It seems this issue is asciidoctor-specific, i.e., it doesn't pertain to the asciidoc tool.

**Comment by mb21** [comment]
So we would have to change this only for `-t asciidoctor` but not `-t asciidoc`?

Potentially related: https://github.com/jgm/pandoc/issues/2337 and https://github.com/jgm/pandoc/issues/4545

**Comment by rwalkerands** [comment]
Oh, I see you're already well aware of this sort of thing ...

Well, I tried the most recent release (8.6.10) of https://github.com/asciidoc/asciidoc and https://github.com/asciidoc/asciidoc-py3 on the output from pandoc, and they both output HTML containing:

```
<div class="paragraph"><p>Test escape {hello} and {} and { } and { a}.</p></div>
```

which seems to be OK.

**Comment by mb21** [comment]
Asciidoc escaping is indeed quite crazy... It would be good to have more extensive tests before we try to fix one escaping case and break two others.

**Comment by rwalkerands** [comment]
Yes.

BTW On the issue I filed against asciidoctor, I was pointed to https://github.com/asciidoctor/kramdown-asciidoc. That program does indeed handle the escapes correctly.

Although it contains a test suite: https://github.com/asciidoctor/kramdown-asciidoc/tree/master/spec/scenarios, I didn't find a set of test cases for escaping.

**Comment by jgm** [comment]
I think this should be closed as a dup of #2337 

Please feel free to add additional examples/test cases to that issue.

In this case, it would help to know why `{hello}` needs an escape but `{}` and `{ a}` don't.  If the rule is simple enough, we may be able to implement it.
